fiduceo_uncertainties.f90

```diff
@@ -784,16 +790,41 @@ SUBROUTINE Write_Temp_NETCDF(temp_file,AVHRR,FCDR,twelve_micron_there)
        call check(stat)
     ENDIF
 
+!MT: 03-11-2017: fix problem of value not filling array     
+!    ALLOCATE(ch1_non_random(nx,ny), stat)
+!    ALLOCATE(ch2_non_random(nx,ny), stat)
+!    ALLOCATE(ch3a_non_random(nx,ny), stat)
+!    IF( stat.ne.0 )THEN
+!       error
+!    ENDIF
+!    ch1_non_random = -1e30
+!    ch2_non_random = -1e30
+!    ch3a_non_random = -1e30
+!    WHERE(ch1_varid.ne.-1e30)ch1_non_random=0.03
+!    WHERE(ch2_varid.ne.-1e30)ch2_non_random=0.05
+!    WHERE(ch3a_varid.ne.-1e30)ch3a_non_random=0.05
+!    DEALLOCATE(ch1_non_random)
+!    DEALLOCATE(ch2_non_random)
+!    DEALLOCATE(ch3a_non_random)
+
 !    WRITE(*,*)'Ch1 (Non-Rand) writing'
-    stat = NF90_PUT_VAR(ncid, ch1_non_random_varid, 0.03)
-    call check(stat)
+    IF( ALLOCATED(AVHRR%new_array1) )THEN
+!       stat = NF90_PUT_VAR(ncid, ch1_non_random_varid, 0.03*AVHRR%new_array1_error)
+       stat = NF90_PUT_VAR(ncid, ch1_non_random_varid, 0.03)
+       call check(stat)
+    ENDIF
 
 !    WRITE(*,*)'Ch2 (Non-Rand) writing'
-    stat = NF90_PUT_VAR(ncid, ch2_non_random_varid, 0.05)
-    call check(stat)
+!MT: 03-11-2017: fix problem of value not filling array     
+    IF( ALLOCATED(AVHRR%new_array2) )THEN
+!       stat = NF90_PUT_VAR(ncid, ch2_non_random_varid, 0.05*AVHRR%new_array2_error)
+       stat = NF90_PUT_VAR(ncid, ch2_non_random_varid, 0.05)
```

Comment:
I thought this way was causing you problems? If it is not, you don't need the code I described above.

Reply:
Correct.

-->  I have commented out these three handling cases